packages/ui/components/ui/slider/index.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import * as SliderPrimitive from '@radix-ui/react-slider';
+import { cn } from '../../../lib/utils';
+import { Thumbs } from './thumbs';
+
+interface BaseProps {
+  /**
+   * Note that this is an array. You can pass more than one value to have
+   * multiple draggable points on the slider.
+   */
+  value: number[];
+  onValueChange: (value: number[]) => void;
+  /**
+   * If set to `true`, will show a tooltip with the current value represented by
+   * the thumb that the user's mouse is over. If set to a function, will call
+   * that function with the current value, then use the returned string as the
+   * tooltip content.
+   */
+  thumbTooltip?: boolean | ((value: number) => string);
+}
+
+// When `segmented` is `true`, the other props are required.
+interface PropsWithSegmented extends BaseProps {
+  /** Segment the slider at each step */
+  segmented: true;
+  min: number;
+  max: number;
+  /** The step size to count by */
+  step: number;
+}
+
+interface PropsWithoutSegmented extends BaseProps {
+  /** Segment the slider at each step */
+  segmented?: false;
+  min?: number;
+  max?: number;
+  /** The step size to count by */
+  step?: number;
+}
+
+/**
+ * Renders a draggable range slider:
+ * |---o-------------|
+ *
+ * @example
+ * ```tsx
+ * <Slider
+ *   min={0}
+ *   max={10}
+ *   step={2}
+ *   value={value}
+ *   onValueChange={onValueChange}
+ *   segmented // to show gaps in the slider for each option
+ * />
+ * ```
+ */
+const Slider = ({
```

Comment:
Git is treating this as a new file, but it's actually moved from `../slider.tsx`, and given some new features:

- The `segmented` prop (see screenshot below)
- The `thumbTooltip` prop for rendering a tooltip over the "thumb" of the slider.

Here's the `<DurationSlider />` when using the `segmented` prop:
![image](https://github.com/penumbra-zone/web/assets/1121544/cab6fdcf-5d49-4bf6-98bd-f12497b21396)


And without the `segmented` prop:

![image](https://github.com/penumbra-zone/web/assets/1121544/65fb1544-ea28-461d-9bc2-3a87d48b9889)